pkg/flare/archive.go

```diff
@@ -66,6 +68,11 @@ func createArchive(zipFilePath string, local bool, confSearchPaths SearchPaths,
 		hostname = "unknown"
 	}
 
+	// create the permissions.log file and write its headers
+	if err := initPermsInfo(tempDir, hostname, os.ModePerm); err != nil {
```

Comment:
I like the interface, but I think I may prefer using a map to store the filename, perms - we could add those with `addPermsInfo` function - and then committing to the file in a single operation as opposed to open/write, etc for every file we add to `permissions.log` what do you think?

Reply:
You're right, I was not so happy with the open/write solution but I didn't want to move an opened `*File` in the `createArchive`, moreover, my first shot was with a dedicated data struct and I didn't choose it because it needs to change some existing func signatures, but probably worth it. I'll implement the changes.